platform/src/main/java/org/stellar/anchor/platform/utils/TransactionHelper.java

```diff
@@ -58,20 +59,31 @@ static GetTransactionResponse toGetTransactionResponse(JdbcSep31Transaction txn)
         .build();
   }
 
-  static GetTransactionResponse toGetTransactionResponse(JdbcSep24Transaction txn) {
+  static GetTransactionResponse toGetTransactionResponse(
+      JdbcSep24Transaction txn, AssetService assetService) {
     Refunds refunds = null;
     if (txn.getRefunds() != null) {
       refunds = toRefunds(txn.getRefunds(), txn.getAmountInAsset());
     }
 
+    String amountInAsset = makeAsset(txn.getAmountInAsset(), assetService, txn);
+    String amountOutAsset = makeAsset(txn.getAmountOutAsset(), assetService, txn);
+    String amountFeeAsset = makeAsset(txn.getAmountFeeAsset(), assetService, txn);
+    String amountExpectedAsset = makeAsset(null, assetService, txn);
+
     return GetTransactionResponse.builder()
         .id(txn.getId())
         .sep(PlatformTransactionData.Sep.SEP_24)
         .kind(PlatformTransactionData.Kind.from(txn.getKind()))
         .status(SepTransactionStatus.from(txn.getStatus()))
-        .amountIn(new Amount(txn.getAmountIn(), txn.getAmountInAsset()))
-        .amountOut(new Amount(txn.getAmountOut(), txn.getAmountOutAsset()))
-        .amountFee(new Amount(txn.getAmountFee(), txn.getAmountFeeAsset()))
+        .amountIn(Amount.create(txn.getAmountIn(), amountInAsset))
+        .amountOut(Amount.create(txn.getAmountOut(), amountOutAsset))
+        .amountFee(Amount.create(txn.getAmountFee(), amountFeeAsset))
+        .amountExpected(new Amount(txn.getAmountExpected(), amountExpectedAsset))
```

Comment:
Is there a reason why some amounts are using `Amount.new()` and others are using `new Amount()`?